src/sentry/release_health/duplex.py

```diff
@@ -801,6 +803,15 @@ def _dispatch_call_inner(
             with timer("releasehealth.sessions.duration", tags=tags, sample_rate=1.0):
                 sessions_result = sessions_fn(*args)
 
+        metrics_result = None
+
+        if should_return_metrics:
+            try:
+                with timer("releasehealth.metrics.duration", tags=tags, sample_rate=1.0):
+                    metrics_result = metrics_fn(*args)
+            except Exception:
+                capture_exception()
```

Comment:
Please double check my reasoning here, but I think we need to run `sessions_fn` in this case, even if `should_check_metrics` is false and `should_return_metrics` is true. Maybe the simplest way to achieve this is by moving this block to before line 802, and then changing the condition in line 802 to `if should_check_metrics or not metrics_result`.

Reply:
that makes sense, I updated the code. I think there's a lot of simplifications one could make to this logic but i don't have the brain for that atm